dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ZenodoHelper.java

```diff
@@ -33,48 +49,201 @@
+     * @return
+     */
+    public static ApiClient createZenodoClient(String zenodoAccessToken) {
+        ApiClient zenodoClient = new ApiClient();
+        // for testing, either 'https://sandbox.zenodo.org/api' or 'https://zenodo.org/api' is the first parameter
+        String zenodoUrlApi = zenodoUrl + "/api";
+        zenodoClient.setBasePath(zenodoUrlApi);
+        zenodoClient.setApiKey(zenodoAccessToken);
+        return zenodoClient;
+    }
+
+    /**
+     * Creates a Zenodo ApiClient using the user's Zenodo token if it exists, throws an exception otherwise.
+     * @param user
+     * @return
+     */
+    public static ApiClient createUserZenodoClient(User user) {
+        Optional<Token> zenodoToken = getZenodoToken(user);
+
+        // Update the zenodo token in case it changed. This handles the case where the token has been changed but an error occurred, so the token in the database was not updated
+        if (zenodoToken.isPresent()) {
+            tokenDAO.update(zenodoToken.get());
+            sessionFactory.getCurrentSession().getTransaction().commit();
+            sessionFactory.getCurrentSession().beginTransaction();
+        } else {
+            LOG.error("{} {}", NO_ZENODO_USER_TOKEN, user.getUsername());
+            throw new CustomWebApplicationException(NO_ZENODO_USER_TOKEN + " " + user.getUsername(), HttpStatus.SC_BAD_REQUEST);
+        }
+
+        final String zenodoAccessToken = zenodoToken.get().getContent();
+        return createZenodoClient(zenodoAccessToken);
+    }
+
+    /**
+     * Creates a Zenodo client using Dockstore's Zenodo access token. Meant to be used with automatic DOI creation.
+     * @return
+     */
+    public static ApiClient createDockstoreZenodoClient() {
+        return createZenodoClient(dockstoreZenodoAccessToken);
+    }
+
+    /**
+     * Get the Zenodo access token and refresh it if necessary
+     *
+     * @param user Dockstore with Zenodo account
+     */
+    public static List<Token> checkOnZenodoToken(User user) {
+        List<Token> tokens = tokenDAO.findZenodoByUserId(user.getId());
+        if (!tokens.isEmpty()) {
+            Token zenodoToken = tokens.get(0);
+
+            // Check that token is an hour old
+            LocalDateTime now = LocalDateTime.now();
+            LocalDateTime updateTime = zenodoToken.getDbUpdateDate().toLocalDateTime();
+            if (now.isAfter(updateTime.plusHours(1).minusMinutes(1))) {
+                LOG.info("Refreshing the Zenodo Token");
+                String refreshUrl = zenodoUrl + "/oauth/token";
+                String payload = "client_id=" + zenodoClientID + "&client_secret=" + zenodoClientSecret
+                        + "&grant_type=refresh_token&refresh_token=" + zenodoToken.getRefreshToken();
+                SourceControlResourceInterface.refreshToken(refreshUrl, zenodoToken, httpClient, tokenDAO, payload);
+            }
+        }
+        return tokenDAO.findByUserId(user.getId());
```

Comment:
This method returns all of the user's tokens. It seems like you could just return the Zenodo token(s).